### ISSUE TITLE
CA1016 raised when AssemblyVersionAttribute defined in a linked file

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn-analyzers#3136]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/462396/ca1016-raised-when-assemblyversionattribute-define.html)._

---
Add a solution file such as VersionInfo.cs
Add an AssemblyVersion attribute to that file
In a project's AssemblyInfo.cs remove any line containing an AssemblyVersion attribute
Add VersionInfo.cs to the project as a linked file

Code analysis incorrectly reports CA1016 Add an AssemblyVersion attribute

NB, this approach is useful if a solution has multiple projects that need to share a version number, copyright information, trademark etc. defined in one place.

---
#### Visual Studio Feedback System on 2/19/2019, 06:07 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Visual Studio Feedback System on 3/15/2019, 06:41 PM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

#### Visual Studio Feedback System on 12/31/2019, 00:03 AM: 

<p>Thank you for sharing your feedback! Our teams prioritize action on product issues with broad customer impact. See details at: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2017#faq">https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2017#faq</a>. In case you need answers to common questions or need assisted support, be sure to use <a target='_blank' href="https://visualstudio.microsoft.com/vs/support/">https://visualstudio.microsoft.com/vs/support/</a>. We’ll keep you posted on any updates to this feedback.</p>

---
### Original Solutions
(no solutions)

**Comment by WayneHiller** [comment]
This warning is bugging me. Is there a way to suppress CA1016? I tried [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1016:MarkAssembliesWithAssemblyVersion")] in the project's AssemblyInfo.cs file but no luck.

**Comment by Evangelink** [comment]
Hey there!

I can't reproduce the issue mentioned before using v2.9.8. Could you confirm this is the version you are using?

FYI I have created a console application, copied the `AssemblyInfo.cs` and renamed it into `VersionInfo.cs`. I have cleaned the 2 files and I don't have any issue.

Content of `VersionInfo.cs`:

```csharp
using System.Reflection;

[assembly: AssemblyVersion("1.0.0.0")]
```

Content of `AssemblyInfo.cs`:

```csharp
using System.Reflection;
using System.Runtime.InteropServices;

// General Information about an assembly is controlled through the following
// set of attributes. Change these attribute values to modify the information
// associated with an assembly.
[assembly: AssemblyTitle("ConsoleApp1")]
[assembly: AssemblyDescription("")]
[assembly: AssemblyConfiguration("")]
[assembly: AssemblyCompany("")]
[assembly: AssemblyProduct("ConsoleApp1")]
[assembly: AssemblyCopyright("Copyright ©  2020")]
[assembly: AssemblyTrademark("")]
[assembly: AssemblyCulture("")]

// Setting ComVisible to false makes the types in this assembly not visible
// to COM components.  If you need to access a type in this assembly from
// COM, set the ComVisible attribute to true on that type.
[assembly: ComVisible(false)]

// The following GUID is for the ID of the typelib if this project is exposed to COM
[assembly: Guid("415871ca-3c54-440e-ab55-c069cebb3451")]

```

**Comment by WayneHiller** [comment]
@Evangelink I updated to version 2.9.8 and the issue seems to be fixed.

**Comment by Evangelink** [comment]
@WayneHiller That's a good news! Thank you for the feedback.

@mavasani I think the ticket can be closed.

**Comment by BillDines** [comment]
Unfortunately I don;t think this is completely fixed in version 2.9.8.

We use a shared .cs file to provide the AssemblyVersionAttribute as described above and are using version 2.9.8 of FXCopAnalysers with the latest VS2019 but still have this problem.  

It is only a small number of projects that have the warning.  There is nothing obviously different about these projects compared to the others.

**Comment by Evangelink** [comment]
@BillDines I have just made a test with 2 console apps, one using the old project format  and the other using the new project format. I have used the exact same code as mentioned earlier and don't have any issue. Would you mind sharing a small reproducer project?

**Comment by LeftofZen** [comment]
this is still an issue. it happens when my csproj file has something like:

```
<Version>13.1.0</Version>
<PackageVersion>$(Version)</PackageVersion>
<AssemblyVersion>$(Version)</AssemblyVersion>
<FileVersion>$(Version)</FileVersion>
```

but not when its 

```
<PackageVersion>13.1.0</PackageVersion>
<AssemblyVersion>$(PackageVersion)</AssemblyVersion>
<FileVersion>$(PackageVersion)</FileVersion>
```

**Comment by rwv37** [comment]
I get this warning a lot, but only sporadically. When it happens, I've found that cargo cultish actions such as rebuilding, or cleaning then rebuilding, or cleaning then closing the IDE then restarting it and reopening the project then rebuilding, eventually make it go away.

So I'm wondering, is it possible that the problem is something like the check for this is done at some nondeterministic time, and occasionally it happens to be done before the assembly has been marked with the version?

**Comment by rwv37** [comment]
By the way, I _think_ this is with VS Community 2022 17.9.2. I say "I think" because I don't remember having gotten this warning since I commented here (last week), and I guess it's possible I updated from an earlier version of VS in the meantime (would have been from a very recent version, almost certainly whatever preceded 17.9.2). I also don't remember doing _that_, but there's a lot of stuff I don't remember nowadays ¯\\\_(ツ)\_/¯
